js/plugins/vertexai/src/gemini.ts

```diff
@@ -732,17 +732,13 @@ function fromGeminiFinishReason(
 }
 
 function toGeminiPart(part: Part): GeminiPart {
-  if (part.text) {
-    return { text: part.text };
-  } else if (part.media) {
-    return toGeminiFileDataPart(part);
-  } else if (part.toolRequest) {
-    return toGeminiToolRequestPart(part);
-  } else if (part.toolResponse) {
-    return toGeminiToolResponsePart(part);
-  } else {
-    throw new Error('unsupported type');
-  }
+  if (part.text) return { text: part.text };
+  if (part.media) return toGeminiFileDataPart(part);
+  if (part.toolRequest) return toGeminiToolRequestPart(part);
+  if (part.toolResponse) return toGeminiToolResponsePart(part);
+  if (typeof part.reasoning === 'string') return toThought(part);
```

Comment:
Shouldn't it follow the naming conventions? i.e. 'toGeminiThoughtPart'?

Reply:
good point, renamed